pandevice/device.py

```diff
@@ -210,3 +212,118 @@ def variables(cls):
             Var("login-banner"),
             Var("update-server"),
         )
+
+
+class PasswordProfile(VersionedPanObject):
+    """Password profile object
+
+    Args:
+        name (str): Password profile name
+        expiration (int): Number of days until the password expires
+        warning (int): Number of days warning before password expires
+        login_count (int): Post expiration admin login count
+        grace_period (int): Post expiration grace period
+
+    """
+    ROOT = Root.MGTCONFIG
+    SUFFIX = ENTRY
+
+    def _setup(self):
+        # xpaths
+        self._xpaths.add_profile(value='/password-profile')
+
+        # params
+        params = []
+
+        params.append(VersionedParamPath(
+            'expiration', vartype='int',
+            path='password-change/expiration-period'))
+        params.append(VersionedParamPath(
+            'warning', vartype='int',
+            path='password-change/expiration-warning-period'))
+        params.append(VersionedParamPath(
+            'login_count', vartype='int',
+            path='password-change/post-expiration-admin-login-count'))
+        params.append(VersionedParamPath(
+            'grace_period', vartype='int',
+            path='password-change/post-expiration-grace-period'))
+
+        self._params = tuple(params)
+
+
+class Administrator(VersionedPanObject):
+    """Administrator object
+
+    Args:
+        name (str): Admin name
+        superuser (bool): Admin type - superuser
+        superuser_read_only (bool): Admin type - superuser, read only
+        panorama_admin (bool): Panonrama - a panorama admin only
+        device_admin (bool): Admin type - device admin
+        device_admin_read_only (bool): Admin type - device admin, read only
+        vsys (list/str): Physical firewalls: the vsys this admin should manage
+        vsys_device (str): The device specification for the vsys admin (default: localhost.localdomain)
+        vsys_read_only (list/str): Physical firewalls: the vsys this read only admin should manage
+        vsys_read_only_device (str): The device specification for the vsys_read_only admin (default: localhost.localdomain)
+        password_hash (encrypted str): The encrypted password
+        password_profile (str): The password profile for this user
```

Comment:
There seems to be some param explosion due to the API schema in this object.  Would love to collapse these if possible, though it might take changes to base.py.  Here's how I envision this as a collapsed set of params:

- name: Admin name
- dynamic_role: Admin role (possible values: superuser, superreader, deviceadmin, devicereader, vsysadmin, vsysreader, panorama_admin)
- admin_role:  Admin role profile for this administrator
- vsys: vsys this admin can manage (conditional on dynamic_role is equal to vsysadmin or vsysreader)
- password_hash
- password_profile

Also, it doesn't look like this supports Admin role profiles yet.